docker/main/docker-build.yml

```diff
@@ -77,6 +79,7 @@ services:
       context: ./ngen
       target: ngen_worker
       args:
+        NUMPY_CONSTRAINT: ${NUMPY_INSTALL_CONSTRAINT:?Please set global NUMPY_INSTALL_CONSTRAINT in environment config}
```

Comment:
This is an abuse of the args parameter. Do not use this to inject python versions. Stick to setup.py and requirements.txt.